salt/metalk8s/addons/logging/loki/deployed/pre-upgrade.sls

```diff
@@ -0,0 +1,101 @@
+{#- Due to a bug in Loki that prevent deletion of the old chunks,
+    Some Loki pods might be stuck in CrashLoopBackOff because the volume is full
+    Run a Job to delete old chunks before the upgrade to restore
+    NOTE: This is automatically handled since 124.1.0
+    So this Logic can be removed in `development/126.0` #}
+
+{%- set loki_sts = salt.metalk8s_kubernetes.get_object(
+    kind="StatefulSet",
+    apiVersion="apps/v1",
+    name="loki",
+    namespace="metalk8s-logging"
+) %}
+{%- if loki_sts and salt.pkg.version_cmp(
+        loki_sts["metadata"]["labels"]["metalk8s.scality.com/version"],
+        "124.1.0"
+    ) == -1 %}
+
+include:
+    - .workaround-job-dep
+
+  {%- from "metalk8s/repo/macro.sls" import build_image_name with context %}
+  {%- set loki_defaults = salt.slsutil.renderer('salt://metalk8s/addons/logging/loki/config/loki.yaml', saltenv=saltenv) %}
+  {%- set loki = salt.metalk8s_service_configuration.get_service_conf('metalk8s-logging', 'metalk8s-loki-config', loki_defaults) %}
+
+  {#- Let's consider we only support retention_period and index period in hours for the moment #}
+  {%- if loki.spec.config.table_manager.retention_period[-1] != "h" %}
+
+Invalid Loki table manager retention_period, should be in hours:
+  test.fail_without_changes: []
+
+  {#- NOTE: Consider the last index is the one currently used #}
+  {%- elif loki.spec.config.schema_config.configs[-1].index.period[-1] != "h" %}
+
+Invalid Loki index period, should be in hours:
+  test.fail_without_changes: []
+
+  {%- else %}
+
+    {#- Let's remove file older than retention time + index time + 3 days #}
+    {%- set max_age = ((
+              (loki.spec.config.table_manager.retention_period[:-1] | int) / 24
+              + (loki.spec.config.schema_config.configs[-1].index.period[:-1] | int) / 24
+            )| int) + 3 %}
+
+Create Loki Cleaner Workaround Job:
+  metalk8s_kubernetes.object_present:
+    - manifest:
+        apiVersion: batch/v1
```

Comment:
The job definition is duplicated, can we easily include it ?

Reply:
similar question for the max age computation logic that we could try to factorize

Reply:
It's not exactly the same since the Loki data dir changed in the latest helm-chart `/data/loki => /var/loki`, but yes it could be factorized using a macro if we want to.

I just think about the fact that since we support 1 major upgrade at a time in MetalK8s, it means we need to keep this workaround until `development/126.0` so in 125.x the Loki definition may change again and may need some update on the `CronJob` but not on the pre-upgrade `Job` since this one will always run on older versions, and for the `max_age` to me the computation might be different between pre-upgrade and the `CronJob` once we will have the boltdb-shipper (but right we could go with a macro for the moment and change it later if it's needed also)

No strong opinion here

Reply:
Done